pkg/frontend/openshiftcluster_listorgetkubernetesobjects_post.go

```diff
@@ -0,0 +1,82 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+	"github.com/Azure/ARO-RP/pkg/util/dynclient"
+	"github.com/Azure/ARO-RP/pkg/util/restconfig"
+)
+
+func (f *frontend) postListOrGetKubernetesObjects(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	body := r.Context().Value(middleware.ContextKeyBody).([]byte)
+	if len(body) > 0 && !json.Valid(body) {
+		api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidRequestContent, "", "The request content was invalid and could not be deserialized.")
+		return
+	}
+
+	r.URL.Path = filepath.Dir(r.URL.Path)
+	// remove the /admin
+	r.URL.Path = strings.Replace(r.URL.Path, "/admin", "", 1)
+	b, err := f._postListOrGetKubernetesObjects(ctx, r)
+
+	reply(log, w, nil, b, err)
+}
+
+func (f *frontend) _postListOrGetKubernetesObjects(ctx context.Context, r *http.Request) ([]byte, error) {
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	vars := mux.Vars(r)
+
+	kinds, ok := r.URL.Query()["kind"]
+	if !ok || len(kinds) == 0 {
+		return nil, api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "", "The request query was invalid: %q.", "kind is required")
+	}
+	if strings.Contains(strings.ToLower(kinds[0]), "secret") {
+		return nil, api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeForbidden, "", "Access to Secrets are forbidden: %q.", "")
+	}
+	namespaces, ok := r.URL.Query()["namespace"]
+	if !ok || len(namespaces) == 0 {
+		namespaces = []string{""}
+	}
+	names, ok := r.URL.Query()["name"]
+	if !ok || len(names) == 0 {
+		names = []string{""}
+	}
+
+	doc, err := f.db.OpenShiftClusters.Get(ctx, strings.Split(r.URL.Path, "?")[0])
+	switch {
+	case cosmosdb.IsErrorStatusCode(err, http.StatusNotFound):
+		return nil, api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s' under resource group '%s' was not found.", vars["resourceType"], vars["resourceName"], vars["resourceGroupName"])
+	case err != nil:
+		return nil, err
+	}
+
+	config, err := restconfig.RestConfig(f.env, doc.OpenShiftCluster)
+	if err != nil {
+		return nil, err
+	}
+
+	dc, err := dynclient.New(log, config)
+	if err != nil {
+		return nil, err
+	}
+	ul, err := dc.GetOrList(ctx, kinds[0], namespaces[0], names[0])
+	if err != nil || ul == nil {
+		return nil, err
+	}
+	return ul.MarshalJSON()
```

Comment:
I am tempted to return yaml to make readability easier - wdyt? @ehashman?

Reply:
Yeah, I'd return YAML unless JSON was the only supported option.